crypto/fipsmodule/rsa/rsa.c

```diff
@@ -230,7 +230,7 @@ void RSA_get0_crt_params(const RSA *rsa, const BIGNUM **out_dmp1,
 
 int RSA_set0_key(RSA *rsa, BIGNUM *n, BIGNUM *e, BIGNUM *d) {
   if ((rsa->n == NULL && n == NULL) ||
-      (rsa->e == NULL && e == NULL)) {
+      (rsa->e == NULL && e == NULL && rsa->d == NULL && d == NULL)) {
```

Comment:
Let's do as before and define a static`detect_stripped_jca_key_something()`, which makes this non-obvious condition more clear...

Reply:
This is an art-not-science opinion, but it seems to me that pulling this out into a separate function would over-complicate things, particularly because we can use the function's doc comment to clarify that lack of `e` is indicative of a JCA key.

Also, in the DER-decoding change from last april, we not only had to detect keys but had to modify them. This was because when we deserialized them, the value of absent fields was `0` instead of `NULL`. Here, the caller has full control of the invocation and can set parameters to `NULL`, so we don't need to (and shouldn't) modify the parameters.

Reply:
i'll update the comment.